CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+  * [Code Quality Standards](#code-quality-standards)
+  * [Security Best Practices](#security-best-practices)
+* [Contribution Types](#contribution-types)
+  * [Adding New Tasks](#adding-new-tasks)
+  * [Modifying Existing Tasks](#modifying-existing-tasks)
+  * [Adding New Pipelines](#adding-new-pipelines)
+  * [Adding External Tasks](#adding-external-tasks)
+  * [Task Deprecation](#task-deprecation)
+* [Review Process](#review-process)
+
+## Project Overview
+
+This repository serves as a central repository for several components used in [Konflux](https://konflux-ci.dev).
+Therefore, everybody should follow its [Code of Conduct](https://github.com/konflux-ci/community/blob/main/CODE_OF_CONDUCT.md).
+
+This central repository contains components managed by the Konflux Build Team and other various contributors.
+See the CODEOWNERS file for more information.
+
+Components are delivered to Konflux via the `konflux-ci/tekton-catalog` Quay organization as OCI bundles.
+
+## How to Report Issues
+
+We encourage early communication for all types of contributions.
+If the contribution is non-trivial (straightforward bugfixes, typos, etc.), please open an issue to discuss your plans and get guidance from maintainers.
+
+### Bug Reports
+
+When reporting bugs, please include the following information:
+
+**Required Information:**
+- **Problem Description**: Clear, concise description of what went wrong
+- **Steps to Reproduce**: Detailed steps to recreate the issue
+- **Current Behavior**: What actually happened
+- **Expected Behavior**: What you expected to happen
+- **Other information**: For example relevant error messages or logs
+
+
+### Feature Requests
+
+For feature requests, please provide:
+
+- **Use Case**: Describe the problem you're trying to solve
+- **Proposed Solution**: Your suggested approach (if any)
+- **Alternatives Considered**: Other solutions you've thought about
+- **Impact**: Who would benefit from this feature
+
+
+## How to Submit Pull Requests
+
+### Development Workflow
+
+1. **Fork and Clone**: Fork this repository and clone your fork
+2. **Create Feature Branch**: Create a new branch from `main`
+3. **Make Changes**: Implement your changes
+4. **Generate Content**: Run generation scripts if needed
+   ```bash
+   ./hack/generate-everything.sh
+   ```
+6. **Commit Changes**: See [commit guidelines](#pull-request-guidelines)
+
+### Pull Request Guidelines
+
+**Commit Requirements:**
+- Write clear, descriptive commit titles. Should fit under 50 characters
+- Write meaningful commit descriptions with each line having less than 72 characters
+- Split your contribution into several commits if applicable, each should represent a logical chunk
+- Add line `Assisted-by: <name-of-ai-tool>` if you used an AI tool for your contribution
+
+**Pull Request Content:**
+- **Title**: Clear, descriptive title. Should fit under 72 characters.
+- **Description**: Explain the changes and their purpose. You may refer to commit descriptions.
```

Comment:
nitpick: I would suggest just a tiny adjustment to reduce the redundancy of information:
_Explain the overall changes and their purpose, this should be a cover letter for your commits_ ...